src/Elastic.Apm/Model/Span.cs

```diff
@@ -145,33 +146,33 @@ internal Span StartSpanInternal(string name, string type, string subType = null,
 			return retVal;
 		}
 
-		public void End(double? duration = null)
+		public void End()
 		{
 			if (Duration.HasValue)
 			{
-				_logger.Debug()?.Log("End() called on already ended {Span}. Start time: {Time} (as timestamp: {Timestamp}), Duration: {Duration}ms",
+				_logger.Trace()?.Log("Ended {Span} (with Duration already set)." +
+					" Start time: {Time} (as timestamp: {Timestamp}), Duration: {Duration}ms",
 					this, TimeUtils.FormatTimestampForLog(Timestamp), Timestamp, Duration);
 			}
 			else
 			{
-				if (duration.HasValue)
-				{
-					Duration = duration.Value;
-					_logger.Trace()?.Log("Ended {Span}. Start time: {Time} (as timestamp: {Timestamp}), Duration: {Duration}ms",
-						this, TimeUtils.FormatTimestampForLog(Timestamp), Timestamp, Duration);
-				}
-				else
-				{
-					var endTimestamp = TimeUtils.TimestampNow();
-					Duration = TimeUtils.DurationBetweenTimestamps(Timestamp, endTimestamp);
-					_logger.Trace()?.Log("Ended {Span}. Start time: {Time} (as timestamp: {Timestamp})," +
-						" End time: {Time} (as timestamp: {Timestamp}), Duration: {Duration}ms",
-						this, TimeUtils.FormatTimestampForLog(Timestamp), Timestamp,
-						TimeUtils.FormatTimestampForLog(endTimestamp), endTimestamp, Duration);
-				}
+				Assertion.IfEnabled?.That(! _dbgIsEnded, $"If a span doesn't have Duration set it means that the span did not end yet." +
+					$" this: {this}; {nameof(Duration)}: {Duration}, {nameof(_dbgIsEnded)}: {_dbgIsEnded}");
+
+				var endTimestamp = TimeUtils.TimestampNow();
+				Duration = TimeUtils.DurationBetweenTimestamps(Timestamp, endTimestamp);
+				_logger.Trace()?.Log("Ended {Span}. Start time: {Time} (as timestamp: {Timestamp})," +
+					" End time: {Time} (as timestamp: {Timestamp}), Duration: {Duration}ms",
+					this, TimeUtils.FormatTimestampForLog(Timestamp), Timestamp,
+					TimeUtils.FormatTimestampForLog(endTimestamp), endTimestamp, Duration);
 			}
 
-			if (IsSampled) _payloadSender.QueueSpan(this);
+			var isFirstEndCall = ! _dbgIsEnded;
+			if (IsSampled && isFirstEndCall)
```

Comment:
nit: I think according the `.editorconfig` we should invert this (not error though). (although I personally find this version more readable)

Reply:
This code had a bug (`_dbgIsEnded` was set to `true` only when `IsSampled` is `true`) so I fixed it and now there is no suggestion from Rider to invert `if`. There was a related bug in `Transaction.End()` (`End()` queued transaction on each call instead of only the 1st one). I fixed both issues and added a test (`Elastic.Apm.Tests.ApiTests.ApiTests.CallingEndMoreThanOnceIsNoop`) to verify.

BTW is it indeed settings in `.editorconfig` that prompt Rider's suggestion to invert that `if` or maybe it's just default Rider's inspections settings?